File: Difficulty: Medium/Sort a stack/sort-a-stack.cpp
class Solution {
  public:
    void sortStack(stack<int> &st) {
        // code here
        if(!st.empty()){
            int temp = st.top();
            st.pop();
            sortStack(st);
            helper(st,temp);
        }
        return;
    }
    void helper(stack<int>& st, int temp){
        if(st.empty() || st.top() <= temp){
            st.push(temp);
            return;
        }
        int num = st.top();
        st.pop();
        helper(st,temp);
        st.push(num);
        return;
    }
};

File: Difficulty: Medium/Sort a stack/sort-a-stack.java
class Solution {
    private void helper(Stack<Integer> st, int temp){
        if(st.empty() || st.peek()<=temp){
            st.push(temp);
            return;
        }
        int val = st.peek();
        st.pop();
        helper(st,temp);
        st.push(val);
        return;
    }
    public void sortStack(Stack<Integer> st) {
        // code here
        while(st.empty()){
            return;
        }
        int temp = st.peek();
        st.pop();
        sortStack(st);
        helper(st,temp);
        return;
    }
}